src/Disks/IDiskRemote.h

```diff
@@ -49,34 +55,50 @@ friend class DiskRemoteReservation;
         const String & log_name_,
         size_t thread_pool_size);
 
-    struct Metadata;
-
-    const String & getName() const final override { return name; }
+    /// Methods to manage local metadata of remote FS objects.
 
-    const String & getPath() const final override { return metadata_path; }
+    struct Metadata;
 
     Metadata readMeta(const String & path) const;
 
     Metadata createMeta(const String & path) const;
 
     Metadata readOrCreateMetaForWriting(const String & path, WriteMode mode);
 
-    UInt64 getTotalSpace() const override { return std::numeric_limits<UInt64>::max(); }
+    /// Disk info
+
+    const String & getName() const final override { return name; }
+
+    const String & getPath() const final override { return metadata_path; }
+
+    UInt64 getTotalSpace() const final override { return std::numeric_limits<UInt64>::max(); }
 
-    UInt64 getAvailableSpace() const override { return std::numeric_limits<UInt64>::max(); }
+    UInt64 getAvailableSpace() const final override { return std::numeric_limits<UInt64>::max(); }
 
-    UInt64 getUnreservedSpace() const override { return std::numeric_limits<UInt64>::max(); }
+    UInt64 getUnreservedSpace() const final override { return std::numeric_limits<UInt64>::max(); }
 
-    UInt64 getKeepingFreeSpace() const override { return 0; }
+    /// Read-only part
 
     bool exists(const String & path) const override;
 
     bool isFile(const String & path) const override;
 
-    void createFile(const String & path) override;
-
     size_t getFileSize(const String & path) const override;
 
+    void listFiles(const String & path, std::vector<String> & file_names) override;
```

Comment:
How do we list files?

Reply:
Same as s3/hdfs - by iterating local metadata.